payments_py/api/observability_api.py

```diff
@@ -0,0 +1,499 @@
+            'accepted_prediction_tokens': 0,
+            'rejected_prediction_tokens': 0,
+        },
+    )
+
+
+def calculate_dummy_song_usage() -> UsageDetails:
+    """Helper function to calculate usage for dummy song operations"""
+    return calculate_song_usage(6)  # Default dummy token count
+
+
+def with_helicone_langchain(
+    model: str,
+    api_key: str,
+    helicone_api_key: str,
+    helicone_base_logging_url: str,
+    account_address: str,
+    environment_name: str,
+    start_agent_request: StartAgentRequest,
+    custom_properties: CustomProperties,
+) -> ChatOpenAIConfiguration:
+    """
+    Creates a ChatOpenAI configuration with Helicone logging enabled
+    """
+    default_headers = get_default_helicone_headers(
+        helicone_api_key,
+        account_address,
+        environment_name,
+        start_agent_request,
+        custom_properties,
+    )
+
+    return ChatOpenAIConfiguration(
+        model=model,
+        api_key=api_key,
+        configuration={
+            'base_url': helicone_base_logging_url,
+            'default_headers': default_headers,
+        }
+    )
+
+
+def with_helicone_openai(
+    api_key: str,
+    helicone_api_key: str,
+    helicone_base_logging_url: str,
+    account_address: str,
+    environment_name: str,
+    start_agent_request: Union[StartAgentRequest, Dict[str, Any]],
+    custom_properties: CustomProperties,
+) -> OpenAIConfiguration:
+    """
+    Creates an OpenAI client configuration with Helicone logging enabled
+    """
+    default_headers = get_default_helicone_headers(
+        helicone_api_key,
+        account_address,
+        environment_name,
+        start_agent_request,
+        custom_properties,
+    )
+
+    return OpenAIConfiguration(
+        api_key=api_key,
+        base_url=helicone_base_logging_url,
+        default_headers=default_headers,
+    )
+
+
+class ObservabilityAPI(BasePaymentsAPI):
+    """
+    The ObservabilityAPI class provides methods to wrap API calls with Helicone logging
+    """
+
+    def __init__(self, options: PaymentOptions):
+        super().__init__(options)
+
+        # TODO: For testing purposes only. Remove once helicone is deployed to staging
+        # Get Helicone API key from environment variable and override the base class property
+        import os
+        self.helicone_api_key = os.getenv('HELICONE_API_KEY') or self.helicone_api_key
+
+        self.helicone_base_logging_url = urljoin(
+            self.environment.helicone_url,
+            'jawn/v1/gateway/oai/v1'
+        )
+        self.helicone_manual_logging_url = urljoin(
+            self.environment.helicone_url,
+            'jawn/v1/trace/custom/v1/log'
+        )
+
+    @classmethod
+    def get_instance(cls, options: PaymentOptions) -> 'ObservabilityAPI':
+        """
+        This method is used to create a singleton instance of the ObservabilityAPI class.
+        """
+        return cls(options)
+
+    async def with_helicone_logging(
```

Comment:
let's remove the reference to helicone in the api methods. maybe this can be renamed to `with_manual_logging`